curvefs/src/client/client_operator.cpp

```diff
@@ -137,7 +138,18 @@ CURVEFS_ERROR RenameOperator::Precheck() {
         return CURVEFS_ERROR::OK;
     } else if (rc == CURVEFS_ERROR::OK) {
         oldInodeId_ = dstDentry_.inodeid();
-        return CheckOverwrite();
+        auto trc = CheckOverwrite();
+        if (trc == CURVEFS_ERROR::OK) {
+            std::shared_ptr<InodeWrapper> inodeWrapper;
+            trc = inodeManager_->GetInode(oldInodeId_, inodeWrapper);
+            if (trc == CURVEFS_ERROR::OK) {
```

Comment:
split it into anther Operator will be better

Reply:
> split it into anther Operator will be better

Yes, can use another op to do, but here only wants to record the oldInode info when overwrite occured, and the precheck also check whether overwrite occured.

Reply:
> split it into anther Operator will be better

agree! maybe add an operator after `Precheck` is better, like:
```
CURVEFS_ERROR RenameOperator::Check4() {
    if (oldInodeId_ == 0) {
        return CURVEFS_ERROR::OK;
    }

    // do something like record `oldInodeSize_` and `oldInodeType_`
}
```

Reply:
agree! maybe add an operator after `Precheck` is better, like:
```
CURVEFS_ERROR RenameOperator::Check4Something() {
    if (oldInodeId_ == 0) {
        return CURVEFS_ERROR::OK;
    }

    // do something like record `oldInodeSize_` and `oldInodeType_`
}
```

Reply:
fixed.